### DISCUSSION TITLE
Is this really a violation of `no-unsafe-finally`?

### DISCUSSION THREAD

**Discussion by jonah-saltzman** [eslint/eslint#16957] [Q&A] [answered]
ESLint is linting on the following (simplified) code for me:

```JS
const obj
try {
   // some async operations
   obj.success = true
} catch {
   // logging operations
   obj.success = false
} finally {
   return obj
}
```
ESLint says this violates `no-unsafe-finally` but in all the example code I see on eslint.org, the unsafety has to do with returning multiple times or throwing inside the catch block, of which my code does neither. Is the above code truly unsafe?

**Comment by Rec0iL99** [comment, marked as answer]
Hi @jonah-saltzman, according to the docs:

> This rule disallows return, throw, break, and continue statements inside finally blocks.

The above code sample is returning a value in the finally block which could be unexpected behaviour. This the rule catches as error.

**Why is this illegal?**

No matter what you return from your try/catch block, the return from the finally block will always be returned thus overwriting the try/catch returns.

If you want to always return something after your try, catch and finally, then you can return at the end of the function which might be something you are trying to do.

```js
async function login(pass) {
   if (pass !== "something") {
      throw new Error("Auth declined")
   }

   return {
      id: 1,
      username: 'Joel'
   }
}

async function foo() {
   const obj = {}
   try {
      // some async operations
      await login("something")
      obj.success = true
   } catch {
      // logging operations
      obj.success = false
   } 

   return obj;
}

await foo()
```

Thanks for using ESLint.
